torchaudio/__init__.py

```diff
@@ -220,149 +184,66 @@ def info(filepath):
-    final output or to save a file with a specific encoding.  For example, one could
-    use the sox ulaw encoding to do 8-bit ulaw encoding.  Note in a tensor output
-    the result will be a 32-bit number, but number of unique values will be determined by
-    the bit precision.
-
-    Returns: sox_encodinginfo_t(object)
-        - encoding (sox_encoding_t), output encoding
-        - bits_per_sample (int), bit precision, same as `precision` in sox_signalinfo_t
-        - compression (float), compression for lossy formats, 0.0 for default compression
-        - reverse_bytes (sox_option_t), reverse bytes, use sox_option_default
-        - reverse_nibbles (sox_option_t), reverse nibbles, use sox_option_default
-        - reverse_bits (sox_option_t), reverse bytes, use sox_option_default
-        - opposite_endian (sox_bool), change endianness, use sox_false
-
-    Example
-        >>> ei = torchaudio.sox_encodinginfo_t()
-        >>> ei.encoding = torchaudio.get_sox_encoding_t(1)
-        >>> ei.bits_per_sample = 16
-        >>> ei.compression = 0
-        >>> ei.reverse_bytes = torchaudio.get_sox_option_t(2)
-        >>> ei.reverse_nibbles = torchaudio.get_sox_option_t(2)
-        >>> ei.reverse_bits = torchaudio.get_sox_option_t(2)
-        >>> ei.opposite_endian = torchaudio.get_sox_bool(0)
-
-    """
-    ei = _torch_sox.sox_encodinginfo_t()
-    sdo = get_sox_option_t(2)  # sox_default_option
-    ei.reverse_bytes = sdo
-    ei.reverse_nibbles = sdo
-    ei.reverse_bits = sdo
-    return ei
-
-
-def get_sox_encoding_t(i=None):
-    r"""Get enum of sox_encoding_t for sox encodings.
-
-    Args:
-        i (int, optional): Choose type or get a dict with all possible options
-            use ``__members__`` to see all options when not specified. (Default: ``None``)
-
-    Returns:
-        sox_encoding_t: A sox_encoding_t type for output encoding
-    """
-    if i is None:
-        # one can see all possible values using the .__members__ attribute
-        return _torch_sox.sox_encoding_t
+        return _soundfile_backend.info(filepath)
     else:
-        return _torch_sox.sox_encoding_t(i)
+        raise NotImplementedError
 
 
-def get_sox_option_t(i=2):
-    r"""Get enum of sox_option_t for sox encodinginfo options.
+def sox_signalinfo_t(*args, **kwargs):
+    warnings.warn(
+        "Please use torchaudio.sox_backend.sox_signalinfo_t instead.",
+        DeprecationWarning,
+    )
+    return sox_backend.sox_signalinfo_t(*args, **kwargs)
 
-    Args:
-        i (int, optional): Choose type or get a dict with all possible options
-            use ``__members__`` to see all options when not specified.
-            (Default: ``sox_option_default`` or ``2``)
-    Returns:
-        sox_option_t: A sox_option_t type
-    """
-    if i is None:
-        return _torch_sox.sox_option_t
-    else:
-        return _torch_sox.sox_option_t(i)
 
+def sox_encodinginfo_t(*args, **kwargs):
```

Comment:
encoding information, signal information etc. are very useful functions in general. we could also think about adding some backend independent interfaces, but maybe after the release.